app/models/sms/incoming.rb

```diff
@@ -21,4 +18,12 @@ def recipient_numbers
   def recipient_hashes(options = {})
     [user: User::SITE, phone: to]
   end
+
+  private
+
+  def set_user
+    matching_users = User.by_phone(from).active
+    self.user ||= matching_users.first if matching_users.count == 1
+    true
+  end
```

Comment:
This is here because it saves a lot of annoyance & some querying in the simplest (most common?) case where we know exactly which user we're fetching going in because there's only one with that number. Adding a comment.